libyul/optimiser/ControlFlowSimplifier.cpp

```diff
@@ -52,7 +52,7 @@ ExpressionStatement makeDiscardCall(
 
 void removeEmptyDefaultFromSwitch(Switch& _switchStmt)
 {
-	boost::remove_erase_if(
+	ranges::actions::remove_if(
```

Comment:
@bshastry If these are semantically different, it rather weird that the tests pass.

Reply:
Hmm... interesting. I am wrong then. May be because I was referring to `std::ranges::remove_if` but the API here is from https://github.com/ericniebler/range-v3/blob/master/include/range/v3/action/remove_if.hpp which iiuc actually calls erase.

Reply:
I am getting lost here, please tell if current fix works, else suggest proposed fix